### ISSUE TITLE
apollo-client 3.8.0 has non-declared dependency on `@types/node-fetch`

### ISSUE THREAD

**Issue by PowerKiKi** [apollographql/apollo-client#11132]
### Issue Description

After upgrading to apollo-client 3.8.0, we started seeing the following error when compiling our typescript projects:

```
node_modules/@apollo/client/utilities/common/responseIterator.d.ts:2:47 - error TS2307: Cannot find module 'node-fetch' or its corresponding type declarations.

2 import type { Response as NodeResponse } from "node-fetch";
~~~~~~~~~~~~


Found 1 error in node_modules/@apollo/client/utilities/common/responseIterator.d.ts:2
```

A possible workaround is to add a dependency on `@types/node-fetch` in our application. But it does not quite feel right. First of all is apollo-client actually meant to depend on some types from node ? if yes, then shouldn't those dependencies declared as such, so that our application does not need to add itself ?

### Link to Reproduction

https://github.com/PowerKiKi/apollo-issue-node-fetch

### Reproduction Steps

```shell
git clone https://github.com/PowerKiKi/apollo-issue-node-fetch && cd apollo-issue-node-fetch && yarn && ./node_modules/.bin/tsc
```

**Comment by jerelmiller** [comment]
Hey @PowerKiKi 👋  

What version were you upgrading from? That responseIterator file was added in v3.7 with the `@defer` work, so I'm a bit surprised to see this pop up now.

**Comment by PowerKiKi** [comment]
`apollo-client` 3.7.17, the version I downgraded to, works without error.

It is a bit suprising indeed. I can build `apollo-angular` without error with `apollo-client` 3.8.0, but I cannot build a project using `apollo-angular` that use `apollo-client` 3.8.0 :thinking: ...

**Comment by jerelmiller** [comment]
Interesting! We'll dig in a little further. Appreciate the reproduction!

**Comment by PowerKiKi** [comment]
I only had a quick glance at the code but it seems there are new exports according to https://github.com/apollographql/apollo-client/commit/42070787ba58ad5b7088a5ce68d9e90b64e6694c#diff-366f17691949b54f007fe0a87b61aea61e42918ab0f7b9ffc5ae91b7307ea5eaR403.

Probably because of https://github.com/apollographql/apollo-client/commit/42070787ba58ad5b7088a5ce68d9e90b64e6694c#diff-4f4a4192cd629d907783a6baf48c530f39d6a43181dd3515cc46231868e0ed08R99. If this was not intentional, maybe revert part of those changes ?

**Comment by Omnired7** [comment]
Am having the same issue, It happens when I upgrade angular from v15 to v16 it doesn't seem to like when I upgrade.

**Comment by jerelmiller** [comment]
Hey @PowerKiKi and @Omnired7

I just opened #11141 which should address this issue. Could you try this snapshot release and check this fixes it?

```
npm i @apollo/client@0.0.0-pr-11141-20230810002434
```

@PowerKiKi I tried this on your reproduction repo and it seems like it builds now, but would love for you to try this out on your own as well to make sure.

**Comment by PowerKiKi** [comment]
I can confirm that `@apollo/client@0.0.0-pr-11141-20230810002434` from #11141 solves the issue both from my reproduction case and my real project (which happens to be Angular too, but that's unrelated).

Thank you for your help !

**Comment by Omnired7** [comment]
works but if I run ng update apollo-angular again after upgrading angular it breaks again.

**Comment by jerelmiller** [comment]
This has just been released in [`3.8.1`](https://github.com/apollographql/apollo-client/releases/tag/v3.8.1). Please upgrade to this version to fix the issue.

@PowerKiKi thanks so much for diving into the commits to find the source of the issue! This saved me quite a bit of time on my end.

**Comment by Omnired7** [comment]
Thank You @jerelmiller
